target/ppc/translate/fixedpoint-impl.c.inc

```diff
@@ -145,22 +145,46 @@ TRANS64(PSTD, do_ldst_PLS_D, false, true, MO_Q)
 
 static bool do_cmp_X(DisasContext *ctx, arg_X_bfl *a, bool s)
 {
-    if (a->l) {
-        REQUIRE_64BIT(ctx);
-        gen_op_cmp(cpu_gpr[a->ra], cpu_gpr[a->rb], s, a->bf);
-    } else {
+    if ((ctx->insns_flags & PPC_64B) == 0) {
```

Comment:
The change seems correct. 2 suggestions:

1. Add the details that are currently in the commit message to comments in the code, to make sure this is known to anyone touching this part of the code (we talked about this recently in our daily meetings - not 100% sure if it was about this exact change, though)
2. Maybe make the code flow easier to understand by separating the handling of the invalid form from the actual op generation, such as:

```
/**
 * (Whole comment about invalid form and L bit)
 */
if (a->l && (ctx->insns_flags & PPC_64B) == 0) {
    if (not booke) {
        qemu_log_mask(...);
    } else {
        return false;
    }
}

if (a->l) {
    gen_op_cmp(...);
} else {
    gen_op_cmp32(...);
}

return true;
```